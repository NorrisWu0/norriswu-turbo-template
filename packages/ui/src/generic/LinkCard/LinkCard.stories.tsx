import { ComponentMeta, ComponentStory } from "@storybook/react";

import { LinkCard } from "./LinkCard";

export default {
  title: "Generic/LinkCard",
  component: LinkCard,
  args: {
    href: "",
    title: "Title",
    description:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Sit fugit non quia perferendis quo facilis autem perspiciatis ad in cumque!",
  },
} as ComponentMeta<typeof LinkCard>;

const Template: ComponentStory<typeof LinkCard> = (args) => (
  <LinkCard {...args} />
);

export const Default = Template.bind({});
