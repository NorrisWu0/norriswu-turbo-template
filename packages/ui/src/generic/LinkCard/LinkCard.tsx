import { Heading, Link, Text } from "@chakra-ui/react";

type LinkCardProps = {
  href: string;
  title: string;
  description: string;
};
export function LinkCard({ href, title, description }: LinkCardProps) {
  const cardStyle = {
    role: "group",
    padding: "1rem 1.2rem",
    borderRadius: "var(--border-radius)",
    background: "rgba(var(--card-rgb), 0)",
    border: "1px solid rgba(var(--card-border-rgb), 0)",
    transition: "background 200ms, border 200ms",
    _hover: {
      background: "rgba(var(--card-rgb), 0.1)",
      border: "1px solid rgba(var(--card-border-rgb), 0.15)",
    },
  };

  const titleStyle = {
    fontSize: "1.5rem",
    fontWeight: 600,
    marginBottom: "0.7rem",
  };

  const arrowStyle = {
    display: "inline-block",
    transition: "transform 200ms",
    _groupHover: {
      transform: "translateX(4px)",
    },
  };

  const descriptionStyle = {
    margin: 0,
    opacity: 0.6,
    fontSize: "0.9rem",
    lineHeight: 1.5,
    maxWidth: "30ch",
  };

  return (
    <Link href={href} target="_blank" rel="noopener noreferrer" {...cardStyle}>
      <Heading as="h2" {...titleStyle}>
        {title}{" "}
        <Text as="span" {...arrowStyle}>
          -&gt;
        </Text>
      </Heading>
      <Text {...descriptionStyle}>{description}</Text>
    </Link>
  );
}
