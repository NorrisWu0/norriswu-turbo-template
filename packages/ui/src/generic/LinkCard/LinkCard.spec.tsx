import { render } from "@testing-library/react";

import { LinkCard } from "./LinkCard";

describe("LinkCard", () => {
  it("should render", () => {
    const { container } = render(
      <LinkCard href="" title="test title" description="test description" />
    );
    expect(container).toMatchSnapshot();
  });
});
