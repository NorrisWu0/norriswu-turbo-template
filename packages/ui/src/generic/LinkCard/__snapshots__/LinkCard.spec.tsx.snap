// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCard should render 1`] = `
<div>
  <a
    class="chakra-link css-kyqrfw"
    href=""
    rel="noopener noreferrer"
    role="group"
    target="_blank"
  >
    <h2
      class="chakra-heading css-5suxjh"
    >
      test title
       
      <span
        class="chakra-text css-1xql7aj"
      >
        -&gt;
      </span>
    </h2>
    <p
      class="chakra-text css-2juig4"
    >
      test description
    </p>
  </a>
</div>
`;
